src/Kernel/ClassDescription.class.st

```diff
@@ -264,6 +264,9 @@ ClassDescription >> classify: selector under: aProtocol [
 
 	"In case the method is already classified and we say to classify it under nil, we do not update the protocol."
 	(aProtocol isNil and: [ oldProtocol isNotNil ]) ifTrue: [ ^ self ].
+	
+	"If we try to classify the method as an extension of its own package, just unclassify the method to avoid any weird state where the method could be considered as an extension or not depending on the interpretation."
+	(self isProtocolExtensionFromTheSamePackage: aProtocol) ifTrue: [ ^ self classify: selector under: Protocol unclassified ].
```

Comment:
Why not also signalling a warning to notify this problem? 
The warning can proceed in interactive mode and is logged in headless mode.

Reply:
I hesitated about that but I agree that it is better. I'll update the PR